zcash_graphs/analysis/analyze.py

```diff
@@ -103,7 +118,7 @@ def extract(self, block, tx):
         else:
             return []
 
-    def aggregate(self, kvs):
+    def aggregate(self, kvs: list[tuple[list[Key], Value]]):
```

Comment:
```suggestion
    def aggregate(self, kvs: list[tuple[list[Key[_]], Value]]) -> XDict[Key[0], Result[0]]:
```
(The `_` in the first `Key` is to indicate that the list actually contains `Key[0], Key[1], Key[2], …`.)


`aggregate` buckets by the first key, then subdivides each of those buckets by
the second key, etc. It then aggregates the smallest buckets using the value
accumulator (the last accumulator in the `bucketers` list), and then passes
the results of that to the previous accumulator, and so on, until we get back
to the accumulator for the first key, and the final result is grouped by that
key.

If we had an `Analysis` with three bucketers, and the accumulators were all the identity function, the specialized
type would look like

``` python
    def aggregate(self, kvs: list[tuple[list[Key[_]], Value]]) -> XDict[Key[0], XDict[Key[1], XDict[Key[2], Value]]]:
```